# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/nsNSSIOLayer.cpp
# Commit: 632eefa7e529
# Full Hash: 632eefa7e529e806bdbecfb50e23f33e2dd86e37
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2021-09-17 21:50:08
# Regressor Bug: 1382886
# File Overlap Count: 1
# Description:
#   Bug 1382886 - Add a callback when the handshake is done. r=necko-reviewers,kershaw,keeler
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123601
# ==============================================================================

diff -r fe6b8f41ca83 -r 632eefa7e529 security/manager/ssl/nsNSSIOLayer.cpp
--- a/security/manager/ssl/nsNSSIOLayer.cpp	Fri Sep 17 13:19:00 2021 +0000
+++ b/security/manager/ssl/nsNSSIOLayer.cpp	Fri Sep 17 13:19:00 2021 +0000
@@ -250,6 +250,10 @@
           ("[%p] nsNSSSocketInfo::SetHandshakeCompleted\n", (void*)mFd));
 
   mIsFullHandshake = false;  // reset for next handshake on this connection
+
+  if (mTlsHandshakeCallback) {
+    Unused << mTlsHandshakeCallback->HandshakeDone();
+  }
 }
 
 void nsNSSSocketInfo::SetNegotiatedNPN(const char* value, uint32_t length) {
@@ -456,6 +460,13 @@
   return NS_OK;
 }
 
+NS_IMETHODIMP
+nsNSSSocketInfo::SetHandshakeCallbackListener(
+    nsITlsHandshakeCallbackListener* callback) {
+  mTlsHandshakeCallback = callback;
+  return NS_OK;
+}
+
 void nsSSLIOLayerHelpers::Cleanup() {
   MutexAutoLock lock(mutex);
   mTLSIntoleranceInfo.Clear();